packages/cosmic-swingset/src/chain-main.js

```diff
@@ -64,6 +64,35 @@ const toNumber = specimen => {
   return number;
 };
 
+/**
+ * A boot message consists of cosmosInitAction fields that are subject to
+ * consensus. See cosmosInitAction in {@link ../../../golang/cosmos/app/app.go}.
+ *
+ * @param {any} initAction
+ */
+const makeBootMsg = initAction => {
+  const {
+    type,
+    blockTime,
+    blockHeight,
+    chainID,
+    isBootstrap,
+    params,
+    supplyCoins,
+    upgradeDetails,
+  } = initAction;
+  return {
+    type,
+    blockTime,
+    blockHeight,
+    chainID,
+    isBootstrap,
+    params,
+    supplyCoins,
+    upgradeDetails,
```

Comment:
I am wondering whether `upgradeDetails` should be included in `bootMsg`. While a chain upgrade may technically bootstrap swingset (hello psm and bulldozer), I'm not sure we have the wiring correct to run any core proposals from the `upgradeDetails` in that case. Regardless, I would not expect the bootstrap vat to take on the responsibility to parse the `upgradeDetails`.